### PR TITLE
#22484 - Customer address States are duplicated in backend.

### PR DESCRIPTION
**Pull request by DmitryPirvu**
- Added condition before addCountryRegions for avoiding duplication of regions in DB.

### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=24770) <br/>All committers have signed the CLA.

File: app/code/Magento/Directory/Setup/Patch/Data/AddDataForAustralia.php [hunk 1]
```diff
@@ -45,12 +49,17 @@ public function __construct(
...
     public function apply()
     {
-        /** @var DataInstaller $dataInstaller */
-        $dataInstaller = $this->dataInstallerFactory->create();
-        $dataInstaller->addCountryRegions(
-            $this->moduleDataSetup->getConnection(),
-            $this->getDataForAustralia()
-        );
+        /** @var \Magento\Directory\Model\ResourceModel\Region\Collection $collection */
+        $collection = $this->regionCollectionFactory->create();
```
Comment (sidolov) [inline app/code/Magento/Directory/Setup/Patch/Data/AddDataForAustralia.php:60]: According to implementation `$this->regionCollectionFactory` is null here and will cause fatal error. Looks like you missed to inject regionCollectionFactory dependency via constructor

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by sidolov** [comment]
Hi @DmitryPirvu  , please, sign CLA, otherwise, we can't process your pull request

File: app/code/Magento/Directory/Setup/Patch/Data/AddDataForAustralia.php [hunk 2]
```diff
@@ -29,28 +30,43 @@ class AddDataForAustralia implements DataPatchInterface, PatchVersionInterface
...
+     *
      * @param ModuleDataSetupInterface $moduleDataSetup
      * @param \Magento\Directory\Setup\DataInstallerFactory $dataInstallerFactory
+     * @param \Magento\Directory\Model\ResourceModel\Region\CollectionFactory $regionCollectionFactory
      */
     public function __construct(
         ModuleDataSetupInterface $moduleDataSetup,
-        \Magento\Directory\Setup\DataInstallerFactory $dataInstallerFactory
+        \Magento\Directory\Setup\DataInstallerFactory $dataInstallerFactory,
+        CollectionFactory $regionCollectionFactory
```
Comment (sidolov) [inline app/code/Magento/Directory/Setup/Patch/Data/AddDataForAustralia.php:47]: Please, inject new dependencies according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by VladimirZaets** [comment]
@DmitryPirvu, I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for collaboration

**Pull request closed without merging**